### PR TITLE
Scripting: Parse stdlib files for parameter names

### PR DESCRIPTION
**Pull request by stu-elastic**
* Task generateContextApiSpec takes optional system parameter
  `jdksrc` with path to extracted java standard library source
  files.
* stdlib source files are the source of `parameter_names` list
  and the `javadoc` value.
* javadoc values may contain newlines and markup such as
  `{@code XX}`, `<p>`, `@throws`

Example method:
```
{
  "declaring": "Appendable",
  "name": "append",
  "return": "Appendable",
  "javadoc": "Appends a subsequence of the specified character sequence to this...",
  "parameters": ["CharSequence", "int", "int" ],
  "parameter_names": ["csq", "start", "end"]
}
```

### CODE REVIEW

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java [hunk 1]
```diff
@@ -52,7 +74,11 @@ public Context(
             List<PainlessContextClassInfo> classInfos = ContextGeneratorCommon.excludeCommonClassInfos(commonClassInfos, info.getClasses());
             classInfos = ContextGeneratorCommon.sortClassInfos(classInfos);
             this.classes = Class.fromInfos(classInfos, javaNamesToDisplayNames);
-            this.importedMethods = Method.fromInfos(info.getImportedMethods(), javaNamesToDisplayNames);
+            this.importedMethods = Method.fromInfos(
+                info.getImportedMethods(),
+                javaNamesToDisplayNames,
+                new StdlibJavadocExtractor.ParsedJavaClass()
```
Comment (stu-elastic) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java:80]: Verify if necessary.

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java [hunk 2]
```diff
@@ -120,21 +168,42 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
...
+        public Method(
+            PainlessContextMethodInfo info,
+            Map<String, String> javaNamesToDisplayNames,
+            String javadoc,
+            List<String> parameterNames
+        ) {
             this.declaring = javaNamesToDisplayNames.get(info.getDeclaring());
             this.name = info.getName();
             this.rtn = ContextGeneratorCommon.getType(javaNamesToDisplayNames, info.getRtn());
+            this.javadoc = javadoc;
```
Comment (jdconrad) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java:186]: Double tab again.

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java [hunk 3]
```diff
@@ -120,21 +168,42 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
...
                 .collect(Collectors.toList());
+            this.parameterNames = parameterNames;
         }
 
-        public static List<Method> fromInfos(List<PainlessContextMethodInfo> infos, Map<String, String> javaNamesToDisplayNames) {
-            return infos.stream()
-                .map(m -> new Method(m, javaNamesToDisplayNames))
-                .collect(Collectors.toList());
+        public static List<Method> fromInfos(
+            List<PainlessContextMethodInfo> infos,
```
Comment (jdconrad) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/PainlessInfoJson.java:194]: Double tab again.

**Review by jdconrad — APPROVED** [review]
LGTM! Thanks for the walkthrough earlier.

A couple of thoughts:
1. We chatted briefly about this, but would you please double check that String is correctly inserted for shared API methods, so we don't end up with the same bug our current documentation has.
2. I wonder if on the design it would make more sense to collect the jdk info after collecting our whitelist info so that if it's null it doesn't have to be checked for null multiple times.

**Comment by stu-elastic** [comment]
master: 7c98d9a
7.x (7.12): 4ab4862

**Pull request merged**